libcxx/utils/formatting/libcxx-pre-commit-formatting-hook.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
```

Comment:
Could we reuse the same script we use in the CI job instead? In `.github/workflows/pr-code-format.yml`, we use `code-format-tools/llvm/utils/git/code-format-helper.py`. If we reused it, it would also solve the problem of formatting our Python files and would ensure that they don't get out of sync.

I would also suggest that this can be done LLVM-wide -- this really shouldn't be libc++ specific since we're doing the same thing as the rest of the monorepo. CC @tru

Reply:
Yeah - I think it should use the python script because of two reasons 1) bash doesn't work on windows 2) we would get the same experience locally and in the CI.

Reply:
Already working on a change to make github actions use this.

I'll make this a little more configurable, but lets not block this on making everyone come with us.

Reply:
No I think it's fine to start with libc++, but I think we should have the discussion and see if the rest of the community would like to follow libc++ here and have the commit hook for everyone. I think it's beneficial.

Reply:
Sure.

Regarding the python script... It doesn't support formatting code outside of github actions. This is meant for our users to use. Ideally we would reconcile the two of these, but I don't think we can make that happen here.

Reply:
> It doesn't support formatting code outside of github actions. This is meant for our users to use. Ideally we would reconcile the two of these, but I don't think we can make that happen here.

Happy to work on this, I rather we fix this now than have two different scripts floating around. I will look at this asap.

Reply:
> It doesn't support formatting code outside of github actions.

What's the issue? It should be possible to change it so that we can use it locally too?

Reply:
Sorry I misunderstood the initial question, so I got a bit confused. These scripts do separate things.

This script should be installed under `.git/hooks/pre-commit`, and it will auto-run whenever the user types `git commit` in their checkout of LLVM.

The ` code-format-tools/llvm/utils/git/code-format-helper.py` is a utility for creating check-runs in a github actions workflow. It also acts as a shallow wrapper around git-clang-format, but it does not run locally nor is it meant for formatting code.

That said, we probably don't need this second github workflow, since these two tools should produce almost the same outcome (ours will format files without extensions that are outside of `libcxx/include`, but the `code-format-helper.py` limits itself to allowing no-extension files under `libcxx/include`.

If we want to fully merge their behaviors, we need to make git-clang-format handle libc++ non-extension headers better. I would be happy taking the same approach `code-format-helper.py` takes 

What am I missing?